### ISSUE TITLE
Single State for Project Visibility Status

### ISSUE THREAD

**Issue by cycomachead** [bromagosa/SnapSite#65]
I think we need to clean up the states for what's public/published. Calling @jmoenig and @brianharvey to this discussion.

The backend has 2 booleans, but there's really only 3 states. 
* Private
* Shared
* Published

In the front-end we currently use 2 icons to communicate state (a link and an eye), each which has a true and a false state. This makes it harder to tell the state of a project at a glance. 

Additionally, changing the state of a project from published to private (or vice-versa) requires 2 clicks.

I propose that in the UI we go to 3 states, each which has a unique icon:

* Private, a closed solid lock
* Shared, perhaps the link, but there's reason for something else.
* Published, an eye -- perhaps a globe

In most places, I think the icon and the text should go together, except for grids. 
In places where all projects are published, I think we could consider dropping the status, but that's a separate issue.

To share a project, I think we should have a modal which lets you pick a setting, and can offer explanatory text. I'm attaching a google drive share modal as one such example. It could also be a customized dropdown menu. 

There's a few reasons we should do this:
* 1 status is much more clear and much less cognitive load to figure out what's going one. 
   * 2 booleans means you have to think about what the combo means to determine what's actually happing
* Groups (Icons + Text) are easier to use
*  All this also is important in terms of general accessibility requirements.

There's two issues still:
* Name of the 3 states. 
* The icon for "shared via link" is naturally a link, but that means we can't use link icon for anything else, like a button that shows you the link to a project or something similar. 

---
<img width="560" alt="Screen Shot 2019-04-17 at 11 20 35 PM" src="https://user-images.githubusercontent.com/1505907/57063264-846aa380-6c77-11e9-8d72-88e900554f87.png">

**Comment by jmoenig** [comment]
I like this proposal, Michael! I agree that the transition from 2 states to 3 often happens by introducing another Boolean, but UI-wise we want to present it as the 3 states we actually want to make available. The implicit technically valid 4th state "NOT shared BUT public" doesn't make sense.

**Comment by bromagosa** [comment]
Good points, Michael. I agree.

Unfortunately, I won't have time to work on this until after the week-long MicroBlocks residency starting on Monday, but I don't believe we're going to release anytime soon anyway...

**Comment by brianharvey** [comment]
I agree too.  The globe for published is good.  Maybe we should steal Google's icon for link sharing, which uses the link but isn't _just_ the link, so it can be distinguished from a link icon alone.  And the lock for private is fine.  And private/shared/published is ok for names, with the advantage that our current users don't have to unlearn "shared" for link sharing.
